src/app/user/user-login/user-login.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, OnInit } from '@angular/core';
+import { UserService } from '../user.service';
+import { NgForm } from '@angular/forms';
+
+@Component({
+  selector: 'app-user-login',
+  templateUrl: './user-login.component.html',
+  styleUrls: ['./user-login.component.css']
+})
+export class UserLoginComponent implements OnInit {
+
+  constructor( private userService:UserService ) { }
+
+  ngOnInit() {
+  }
+
+  // handler for login
+  onSubmit(form: NgForm) {
+  	if(!form.valid) {
+  		return;
+  	}
+
+  	this.userService.login(form.value.username, form.value.password, this.handleSuccessfulLogin, this.handleFailedLogin);
+  }
```

Comment:
Try using notificationService. U dont need 2 send the 2 function. Actually you donr need them at all. In the servixe like we did before, call notifications with the message you got from BE

Reply:
Yeah I wasn't sure how to use notification service.